packages/sdg/src/lib/paginator/paginator.component.ts

```diff
@@ -0,0 +1,167 @@
+import { DOCUMENT, NgFor, NgIf, NgStyle } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Inject,
+  Injector,
+  OnInit,
+  effect,
+  inject,
+  input,
+  output
+} from '@angular/core';
+import { MatButtonModule } from '@angular/material/button';
+import { MatIconModule } from '@angular/material/icon';
+
+import { IgoLanguageModule } from '@igo2/core/language';
+
+@Component({
+  selector: 'sdg-paginator',
+  standalone: true,
+  imports: [
+    NgIf,
+    NgFor,
+    NgStyle,
+    IgoLanguageModule,
+    MatButtonModule,
+    MatIconModule
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  templateUrl: './paginator.component.html',
+  styleUrls: ['./paginator.component.scss']
+})
+export class PaginatorComponent implements OnInit {
+  constructor(@Inject(DOCUMENT) private document: Document) {}
+  injector = inject(Injector);
+
+  listLength = input.required<number>();
+  pageSize = input.required<number>();
+  middlePagesMaxRange = input<number>(1);
+  initialPageIndex = input<number>(0);
+  isHandset = input<boolean>();
+
+  pageChange = output<number>();
+
+  initialPageIndexFirstChange = true;
+
+  currentPageIndex: number = 0;
+  nbOfPages: number = 0;
+
+  firstPagesIndexes: number[] = [];
+  middlePagesIndexes: number[] = [];
+  lastPagesIndexes: number[] = [];
+  pagesIndexes: number[] = [];
```

Comment:
Même ces paramètres auraient pu être convertie en computed (Signal). Il y a surement un niveau de complexité vu qu'il y a beaucoup de manipulation et dépendances. Cela pourrait être un nice to have éventuellement...